File: pascal/concepts/operators.pas
unit Operators;

interface

procedure DemonstrateOperators;

implementation

procedure DemonstrateOperators;
var
  A, B: Integer;
  X, Y: Boolean;
  Result: Real;
begin
  WriteLn;
  WriteLn('=== Operators ===');
  
  A := 10;
  B := 3;
  
  { Arithmetic operators }
  WriteLn('Arithmetic Operators:');
  WriteLn('Addition: ', A + B);
  WriteLn('Subtraction: ', A - B);
  WriteLn('Multiplication: ', A * B);
  Result := A / B;
  WriteLn('Division: ', Result:0:2);
  WriteLn('Integer Division: ', A div B);
  WriteLn('Modulus: ', A mod B);
  
  { Comparison operators }
  WriteLn;
  WriteLn('Comparison Operators:');
  WriteLn('Is A > B? ', A > B);
  WriteLn('Is A = B? ', A = B);
  
  { Logical operators }
  X := True;
  Y := False;
  WriteLn;
  WriteLn('Logical Operators:');
  WriteLn('X AND Y: ', X and Y);
  WriteLn('X OR Y: ', X or Y);
  WriteLn('NOT X: ', not X);
end;

end.

File: dart/lib/concepts/operators.dart
void demonstrateOperators() {
  print('\n=== Operators ===');

  int a = 10;
  int b = 3;

  // Arithmetic operators
  print('Addition: ${a + b}');
  print('Subtraction: ${a - b}');
  print('Multiplication: ${a * b}');
  print('Division: ${a / b}'); // Returns double
  print('Integer Division: ${a ~/ b}'); // Returns int
  print('Modulus: ${a % b}');

  // Comparison operators
  print('\nComparison:');
  print('Is a > b? ${a > b}');
  print('Is a == b? ${a == b}');

  // Logical operators
  bool x = true;
  bool y = false;
  print('\nLogical:');
  print('x AND y: ${x && y}');
  // ignore: dead_code
  print('x OR y: ${x || y}');
  print('NOT x: ${!x}');

  // Increment/Decrement
  int i = 0;
  print('\nIncrement/Decrement:');
  print('i++: ${i++}'); // Post-increment
  print('++i: ${++i}'); // Pre-increment
}
